oe-account-service/op-energy-account-service/src/OpEnergy/BlockTimeStrike/Server/V1/BlockTimeStrikeGuessService.hs

```diff
@@ -260,27 +260,55 @@ getBlockTimeStrikesGuessesPage mpage mfilter = profile "getBlockTimeStrikesGuess
   latestUnconfirmedBlockHeightV <- asks (BlockTime.latestUnconfirmedBlockHeight . blockTimeState)
   mlatestUnconfirmedBlockHeight <- liftIO $ TVar.readTVarIO latestUnconfirmedBlockHeightV
   configBlockTimeStrikeGuessMinimumBlockAheadCurrentTip <- asks (configBlockTimeStrikeGuessMinimumBlockAheadCurrentTip . config)
+  latestConfirmedBlockV <- asks (BlockTime.latestConfirmedBlock . blockTimeState)
+  mlatestConfirmedBlock <- liftIO $ TVar.readTVarIO latestConfirmedBlockV
   recordsPerReply <- asks (configRecordsPerReply . config)
   let
       linesPerPage = maybe recordsPerReply (maybe recordsPerReply id . blockTimeStrikeGuessResultPublicFilterLinesPerPage . fst . unFilterRequest ) mfilter
-  case mlatestUnconfirmedBlockHeight of
-    Nothing -> do
-      let msg = "getBlockTimeStrikesGuessesPage: no confirmed block found yet"
-      runLogging $  $(logError) msg
-      throwJSON err500 msg
-    Just latestUnconfirmedBlockHeight -> do
+  case (mlatestUnconfirmedBlockHeight, mlatestConfirmedBlock) of
+    (Just latestUnconfirmedBlockHeight, Just latestConfirmedBlock) -> do
       let finalFilter =
             case maybe Nothing (blockTimeStrikeGuessResultPublicFilterClass . fst . unFilterRequest) mfilter of
               Nothing -> filter
               Just BlockTimeStrikeFilterClassGuessable ->
                 let
                     minimumGuessableBlock = latestUnconfirmedBlockHeight + naturalFromPositive configBlockTimeStrikeGuessMinimumBlockAheadCurrentTip
+                    minimumGuessableMediantime
+                      = fromIntegral (blockHeaderMediantime latestConfirmedBlock)
+                      + 600
+                        * (naturalFromPositive configBlockTimeStrikeGuessMinimumBlockAheadCurrentTip
+                          + (latestUnconfirmedBlockHeight - blockHeaderHeight latestConfirmedBlock)
+                          )
+                    isStrikeBlockHeightGuessable
+                      = BlockTimeStrikeBlock >=. minimumGuessableBlock
+                    isStrikeMediantimeGuessable
+                      = BlockTimeStrikeStrikeMediantime
+                        >. fromIntegral minimumGuessableMediantime
                 in
-                  ( (BlockTimeStrikeBlock >=. minimumGuessableBlock) -- block height should match threshold
-                  :filter
+                  ( isStrikeBlockHeightGuessable
+                  : isStrikeMediantimeGuessable
+                  : filter
                   )
-              Just BlockTimeStrikeFilterClassOutcomeKnown -> filter
-              Just BlockTimeStrikeFilterClassOutcomeUnknown -> filter
+              Just BlockTimeStrikeFilterClassOutcomeKnown ->
+                let
+                    isStrikeBlockHeightObserved
```

Comment:
First of all "Observed" is hard to understand. (I am possibly guilty of this for using this term first.) 

What do we actually mean? I think we mean something like "Outcome is already known for the strike"

But that's confusing too. 

Do we mean 

A -- the outcome has already been recorded? 
B -- or that the outcome SHOULD have been recorded assuming all the machinery is working right? (But maybe it wasn't recorded) 

If it's A, then all we care about is the strike has a FAST/SLOW record, or equivalently a judgement hash. Much easier to understand the logic. 

if it's B.... I'm not sure. Why should we care about B anyay, for a filter? 

maybe isStrikeBlockHeightObserved is poorly named, or the wrong thing?